File: C_files/Lecture_2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

//Function Declaration
int number_of_letters(string input);
int number_of_words(string input);
int number_of_sentences(string input);

int main()
{
    //Get input from user
    string input = get_string("Text :");

    //Calculate number of letter, word, sentences
    int letter = number_of_letters(input);
    int words = number_of_words(input);
    int sentences = number_of_sentences(input);

    //Calculate Grade
    float L = ((float) letter / (float) words) * 100.0;
    float S = ((float) sentences / (float) words) * 100.0;
    int grade = round(0.0588 * L - 0.296 * S - 15.8);

    //PRINT RESULTS
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }

    return 0;
}

//Count number of letters
int number_of_letters(string input)
{
    int count = 0;
    for (int i = 0; i < strlen(input); i++)
    {
        if (isalpha(input[i]))
        {
            count++;
        }
    }
    return count;
}

//Count number of words
int number_of_words(string input)
{
    int count = 1;
    for (int i = 0; i < strlen(input); i++)
    {
        if (input[i] == ' ')
        {
            count++;
        }
    }
    return count;
}

//Count number of sentences
int number_of_sentences(string input)
{
    int count = 0;
    for (int i = 0; i < strlen(input); i++)
    {
        if (input[i] == '.' || input[i] == '?' || input[i] == '!')
        {
            count++;
        }
    }
    return count;
}

File: Python_files/Lecture_6/sentimental-readability/readability.py
# Number of letters function
def num_of_letters(input):
    count = 0
    for char in input:
        if char.isalpha():
            count += 1
    return count

# Number of words function
def num_of_words(input):
    count = 1
    for char in input:
        if char == " ":
            count += 1
    return count

# Number of sentences function
def num_of_sentences(input):
    count = 0
    for char in input:
        if char in [".", "!", "?"]:
            count += 1
    return count
        

def main():
    # Get user input
    input_value = input("Text: ")

    # Calculate variables
    letter = num_of_letters(input_value)
    words = num_of_words(input_value)
    sentences = num_of_sentences(input_value)
    L = (letter / words) * 100
    S = (sentences / words) * 100

    # Calculate grade
    grade = round(0.0588 * L - 0.296 * S - 15.8)

    # Print results
    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")

if __name__ == "__main__":
    main()

